infra/nnfw/python/setup.py

```diff
@@ -69,10 +70,15 @@
         os.makedirs(arch_path)
         print(f"Created directory '{arch_path}'")
 
-        product_dir = os.path.join(THIS_FILE_DIR, '../../../Product')
+        def get_directories():
+            # If the environment variable is not set, get default one.
+            product_dir = os.environ.get("PRODUCT_DIR", DEFAULT_PRODUCT_DIR)
+            return os.path.join(THIS_FILE_DIR, product_dir), os.path.join(
+                product_dir, "lib/" if product_dir != DEFAULT_PRODUCT_DIR else
+                target_arch + '-linux.release/out/lib')
```

Comment:
I think python wheel may be created in debug build. Also, though we did not checked, on Tizen build, it has `-tizen.xxx`.

Reply:
IMO, This script will be used for python packaging, so default packaging path can be release directory. And this PR support environment variable setting to overwrite path, then we can also try debug build packaging.

Reply:
@hseok-oh @glistening thank you for pointing it out! I am really fresh to the project and still learning about it.

What would you see here as a better way of handling it? Maybe some regex to correctly capture the libs?
The second concern would be that user create both debug and release, what is the priority? For example, user did not cleaned up `Product` directory between builds.
Should there be a notification about it? Or this should also be configurable with some flag?
I am referring to such scenario:
```shell
~/Projects/ONE$ ls Product/
buildtool  out  x86_64-linux.debug  x86_64-linux.release
```
Taking it all into consideration, maybe the best approach would be to use`PRODUCT_DIR`and `CORE_DIR` pair.
That will allow to select the specific `product_dir` and `so_core_dir` in cases where multiple builds are created.
My proposal for the rules will be following:

* Setting `PRODUCT_DIR` will follow path of appending `lib` as builds with `./nnfw configure -DCMAKE_BUILD_TYPE=Release/Debug -DCMAKE_INSTALL_PREFIX="/path/to/install" will not create separate directories by default.
* Setting both `PRODUCT_DIR` and `CORE_DIR` will throw warning that `CORE_DIR` is discarded due to using external installation directory.
* If multiple `arch-system.build-type` exists in Product folder, users will be prompted to pick one/forced to specify with `CORE_DIR`(? prompting is not great for CI work).
* If there is only one installation found or both `PRODUCT_DIR` and `CORE_DIR` are not specified, that will go into search with regex described above.

**The second approach would be to look out for `Product/out` directory instead of `Product/arch-system.type`. That will always be the most recent build by user.**
It's way easier to maintain such path and also is direct link to the installation folder. So the code would look like this:
```python
# Example why this way it would work
# lrwxrwxrwx  1 user user   62 Jul 18 10:45 out -> /home/user/Projects/ONE/Product/x86_64-linux.debug/out
        def get_directories():
            # If the environment variable is not set, get default one.
            product_dir = os.environ.get("PRODUCT_DIR", DEFAULT_PRODUCT_DIR)
            return os.path.join(THIS_FILE_DIR, product_dir), os.path.join(
                product_dir, "lib/" if product_dir != DEFAULT_PRODUCT_DIR else 'out/lib')
```
If that was up to me, I would pick this option due to straightforwardness (always the most recent build) and maintainability of it.

What do you think about such solutions? Which one do you prefer?

Reply:
I think this script will be rarely used by user. It may be used on CI infra for package generation and upload to package repository. So I think this PR is enough.

Reply:
@glistening thank you for the feedback. I will then proceed with incremental improvements of API.